src/neuroglancer_scripts/accessor.py

```diff
@@ -35,15 +36,51 @@ def get_accessor_for_url(url, accessor_options={}):
     r = urllib.parse.urlsplit(url)
     if r.scheme in ("", "file"):
         from neuroglancer_scripts import file_accessor
+        from neuroglancer_scripts import sharded_base
         flat = accessor_options.get("flat", False)
         gzip = accessor_options.get("gzip", True)
         compresslevel = accessor_options.get("compresslevel", 9)
         pathname = _convert_split_file_url_to_pathname(r)
-        return file_accessor.FileAccessor(pathname, flat=flat, gzip=gzip,
-                                          compresslevel=compresslevel)
+
+        accessor = file_accessor.FileAccessor(pathname, flat=flat, gzip=gzip,
+                                              compresslevel=compresslevel)
+        is_sharding = False
+        if accessor_options.get("sharding"):
+            is_sharding = True
+        if not is_sharding:
+            try:
+                info = json.loads(accessor.fetch_file("info"))
+                if sharded_base.ShardedAccessorBase.info_is_sharded(info):
+                    is_sharding = True
+            except Exception:
```

Comment:
Do we really need to be as broad as `Exception` here? Maybe `IOError` would be specific enough? We don't want that statement to catch other kinds of exceptions, like e.g. malformed info

Reply:
Agreed. 

I would like to catch `DataAccessError`, it seems. 

I will fix this =)